rust/loqui_client/src/client.rs

```diff
@@ -0,0 +1,46 @@
+use crate::connection_handler::{ConnectionHandler, InternalEvent};
+use crate::Config;
+use failure::Error;
+use futures::sync::oneshot;
+use loqui_connection::{Encoder, LoquiError, Supervisor as SupervisedConnection};
+use std::net::SocketAddr;
+use std::sync::Arc;
+use tokio::await;
+
+#[derive(Clone)]
+pub struct Client<E: Encoder> {
+    connection: Arc<SupervisedConnection<ConnectionHandler<E>>>,
+}
+
+impl<E: Encoder> Client<E> {
+    pub async fn connect<A: AsRef<str>>(
```

Comment:
Connect returns a connection object immediately - maybe it should be more aptly named `new` or something? To me (following say BSD sockets), `connect` generally returns when the connection is established.

Reply:
Good point, this got lost in a code shift around. I think it should actually just wait until the connection is ready. What do you think?

Reply:
As in it returns a future when conn is ready? That'd work!